internal/service/alarms/internal/alertmanager/alertmanager.go

```diff
@@ -61,3 +66,103 @@ func Setup(ctx context.Context, cl client.Client) error {
 	slog.Info("Successfully configured alertmanager")
 	return nil
 }
+
+// ConvertAmToAlarmEventRecordModels get alarmEventRecords based on the alertmanager notification and AlarmDefinition
+func ConvertAmToAlarmEventRecordModels(am *api.AlertmanagerNotification, aDefinitionRecords []models.AlarmDefinition) []models.AlarmEventRecord {
+	records := make([]models.AlarmEventRecord, 0, len(am.Alerts))
+	for _, alert := range am.Alerts {
+		slog.Info("Converting Alertmanager alert", "alert name", GetAlertName(*alert.Labels))
+		record := models.AlarmEventRecord{
+			AlarmRaisedTime:   *alert.StartsAt,
+			AlarmClearedTime:  alert.EndsAt,
+			PerceivedSeverity: getPerceivedSeverity(*alert.Labels),
+			AlarmStatus:       string(*alert.Status),
+			Fingerprint:       *alert.Fingerprint,
+			ResourceID:        GetResourceID(*alert.Labels),
+			ResourceTypeID:    GetResourceTypeID(*alert.Labels),
+		}
+
+		// Update Extensions with things we didn't really process
+		record.Extensions = getExtensions(*alert.Labels, *alert.Annotations)
+
+		// See if possible to pick up additional info from its definition
+		for _, def := range aDefinitionRecords {
+			if def.AlarmName == GetAlertName(*alert.Labels) { // TODO: match with the resourceTypeID too once init data is ready
+				record.AlarmDefinitionID = def.AlarmDefinitionID
+				record.ProbableCauseID = def.ProbableCauseID
+			}
+		}
+
+		// Note: note only thing we are missing is `AlarmChangedTime` since it's not available from alertmanager
```

Comment:
as far as I could see, EndsAt is different than 0000.... when the alert is resolved, changed time must have this value since changedtime is updated whenever any other field changes

Reply:
Basically there's no specific "changesAt" time in alertmanager (unlike StartsAt or EndsAt...couldnt find it in AM project about it either). So instead of inferring something (e.g keep track of changes in the row) we are letting it stay null. Touched base with Brent about this and he said to let go through as null.

Reply:
I know, but a update trigger might do the trick, it can check the values for us

Reply:
Yeah that's easy to do (i originally had this tbh) but again...given its not coming from alertmanager i was told just keep it as null. But we can simply do that from table so no code changes needed...I'll take that on once im back from PTO as would need test out.